File: 1022 - Sum of Root To Leaf binary Numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumRootToLeaf = function (root) {
    return traverse(root, 0);
};

function traverse(root, s) {
    s = Math.floor(s * 2) + root.val;
    let l = 0, r = 0;
    if (root.left == null && root.right == null) {
        return s;
    }
    if (root.left != null) {
        l = traverse(root.left, s);
    }
    if (root.right != null) {
        r = traverse(root.right, s);
    }
    return l + r;
}

File: 1022 - Sum of Root To Leaf binary Numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptrptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    int traverse(TreeNode* root, int s) {
        s = s * 2 + root->val;
        int l = 0, r = 0;
        if (root->left == nullptr && root->right == nullptr) {
            return s;
        }
        if (root->left != nullptr) {
            l = traverse(root->left, s);
        }
        if (root->right != nullptr) {
            r = traverse(root->right, s);
        }
        return l + r;
    }
public:
    int sumRootToLeaf(TreeNode* root) {
        return traverse(root, 0);
    }
};